src/origyn_nft_reference/metadata.mo

```diff
@@ -1238,10 +1238,51 @@ module {
     };
 
     SB.add(ledger, newTrx);
+    
+    // Call for the annouce function
+    announceTransaction(state, rec, caller, newTrx);
 
     return #ok(newTrx);
   };
 
+  public func announceTransaction(state : Types.State, rec : Types.TransactionRecord, caller : Principal, newTrx : Types.TransactionRecord) : async Types.TransactionRecord {
+        let subscriberId = caller;
+        let eventName = "transaction";
+
+        // Get the previous subscription from the events repo
+        let prevSubscriptionInfo = Info.getSubscriberInfo(caller, state, (subscriberId, eventName));
+
+        // Check if the subscriber is subscribed to the transaction event
+        let result = switch (prevSubscriptionInfo) {
```

Comment:
Our nft should not have to worry about subscriptions.  It should only need to initialize the event system actor and publish an event.  @ZhenyaUsenko has a client that he is working on where you should just be able to pass the ID of the Main event handler and it will take care of the rest.  You guys please work together to get this implemented.